org.opentosca.container.core/src/org/opentosca/container/core/next/repository/NodeTemplateInstanceRepository.java

```diff
@@ -35,4 +35,21 @@ public NodeTemplateInstanceRepository() {
             return q.getResultList();
         }
     }
+
+    public Collection<NodeTemplateInstance> findByTemplateType(final QName templateType) {
+        try (AutoCloseableEntityManager em = EntityManagerProvider.createEntityManager()) {
+            final CriteriaBuilder cb = em.getCriteriaBuilder();
+            // Parameters
+            final ParameterExpression<QName> templateTypeParameter = cb.parameter(QName.class);
+            // Build the Criteria Query
+            final CriteriaQuery<NodeTemplateInstance> cq = cb.createQuery(NodeTemplateInstance.class);
+            final Root<NodeTemplateInstance> sti = cq.from(NodeTemplateInstance.class);
+            cq.select(sti).where(cb.equal(sti.get("templateType"), templateTypeParameter));
+            // Create a TypedQuery
+            final TypedQuery<NodeTemplateInstance> q = em.createQuery(cq);
+            q.setParameter(templateTypeParameter, templateType);
+            // Execute
+            return q.getResultList();
```

Comment:
the comments in this method are IMO mostly useless apart from the vertical space they provide to separate different parts